torchdata/datapipes/iter/util/common.py

```diff
@@ -0,0 +1,112 @@
+# Copyright (c) Meta Platforms, Inc. and affiliates.
+# All rights reserved.
+#
+# This source code is licensed under the BSD-style license found in the
+# LICENSE file in the root directory of this source tree.
+
+import copy
+
+from typing import Callable, Iterable, Iterator, TypeVar
+
+from torch.utils.data.datapipes.utils.common import _check_unpickable_fn
+from torchdata.datapipes import functional_datapipe
+
+from torchdata.datapipes.iter import IterDataPipe, Mapper
+
+T_co = TypeVar("T_co", covariant=True)
+
+
+class MapTemplateIterDataPipe(Mapper[T_co]):
+    def __init__(self, source_datapipe, input_col=None, output_col=None) -> None:
+        fn = self._map
+        super().__init__(source_datapipe, fn=fn, input_col=input_col, output_col=output_col)
+
+    def _map(self, *args, **kwargs) -> T_co:
+        raise NotImplementedError
+
+
+@functional_datapipe("flatmap_proto")
+class FlatMapperProtoIterDataPipe(IterDataPipe[T_co]):
```

Comment:
Do you want to directly change the `FlatMapper` here?
https://github.com/pytorch/data/blob/14b2291f0f0488632e9689b9735c58a3b113d237/torchdata/datapipes/iter/transform/callable.py#L93-L94

Reply:
Yes I do, was making sure I cover all use cases before replacement.